processing/src/main/java/io/druid/segment/data/RangeIndexedInts.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.google.common.base.Preconditions;
+import it.unimi.dsi.fastutil.ints.IntIterator;
+import it.unimi.dsi.fastutil.ints.IntIterators;
+
+import java.io.IOException;
+
+/**
+ * An IndexedInts that always returns [0, 1, ..., N].
+ */
+public class RangeIndexedInts implements IndexedInts
+{
+  private static final int CACHE_LIMIT = 8;
+  private static final RangeIndexedInts[] CACHE = new RangeIndexedInts[CACHE_LIMIT];
+
+  static {
+    for (int i = 0; i < CACHE_LIMIT; i++) {
+      CACHE[i] = new RangeIndexedInts(i);
+    }
+  }
+
+  private final int size;
+
+  private RangeIndexedInts(int size)
+  {
+    this.size = size;
+  }
+
+  public static RangeIndexedInts create(final int size)
```

Comment:
Maybe call this method "acquire()" or "getOrCreate()", because it doesn't necessarily create something?